common/lib/xmodule/xmodule/course_metadata_utils.py

```diff
@@ -0,0 +1,188 @@
+"""Simple utility functions that operate on course metadata.
+
+This is a place to put simple functions that operate on course metadata. It
+allows us to share code between the CourseDescriptor and CourseOverview
+classes, which both need these type of functions.
+"""
+from datetime import datetime
+from base64 import b32encode
+
+from django.utils.timezone import UTC
+
+from .fields import Date
+
+DEFAULT_START_DATE = datetime(2030, 1, 1, tzinfo=UTC())
+
+
+def clean_course_key(course_key, padding_char):
+    """
+    Encode a course's key into a unique, deterministic base32-encoded ID for the course.
+
+    Arguments:
+        course_key (CourseKey): A course key.
+        padding_char (str): Character used for padding at end of the encoded string.
+                            The standard value for this is '='.
+    """
+    return "course_{}".format(
+        b32encode(unicode(course_key)).replace('=', padding_char)
+    )
+
+
+def url_name_for_course_location(location):
+    """
+    Given a course's location, returns the course's URL name.
+
+    Arguments:
+        location (UsageKey): The course's location.
+    """
+    return location.name
+
+
+def display_name_with_default(course):
+    """
+    Calculates the display name for a course.
+
+    Default to the display_name if it isn't None, else fall back to creating
+    a name based on the URL.
+
+    Unlike the rest of this module's functions, this function takes an entire
+    course descriptor/overview as a parameter. This is because certain test
+    cases (specifically, {Text|Image|Video}AnnotationModuleTestCase.test_student_view)
+    create scenarios where course.display_name is not None but course.location
+    is None, which causes calling course.url_name to fail. So, although we'd
+    like to just pass course.display_name and course.url_name as arguments to
+    this function, we can't do so without breaking those tests.
+
+    Arguments:
+        course (CourseDescriptor|CourseOverview): descriptor or overview of said course.
+    """
+    return (
+        course.display_name if course.display_name is not None
+        else course.url_name.replace('_', ' ')
+    ).replace('<', '&lt;').replace('>', '&gt;')
+
```

Comment:
Since this method takes a class *and* operates only on the class members, it really wants to be a class method taking a `self` parameter. Could these methods possibly be a mix-in included by both classes?